### PR TITLE
[Workload Server] move workflow creation out of distribution module

### PR DESCRIPTION
**Pull request by rhuffy**
## General
**Before this PR**:
`atlas-workload-server-distribution` includes workflow creation.

**After this PR**:
This refactor allows closed-source distributions of the workload server to duplicate less code, as it moves the workload creation logic to a shared module, and isolates all Dropwizard-specific components to the distribution module.

==COMMIT_MSG==
[Workload Server] move workflow creation out of distribution module
==COMMIT_MSG==

**Please tag any other people who should be aware of this PR**:
@jeremyk-91
@jkozlowski

### CODE REVIEW

**Review by jkozlowski — DISMISSED** [review]
lgtm, let's wait for Jeremy to have a look

File: atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/config/WorkloadServerInstallConfiguration.java [hunk 1]
```diff
@@ -28,13 +26,7 @@
...
 import com.palantir.atlasdb.workload.workflow.bank.BankBalanceWorkflowConfiguration;
 import com.palantir.atlasdb.workload.workflow.ring.RingWorkflowConfiguration;
-import com.palantir.logsafe.DoNotLog;
-import org.immutables.value.Value;
 
-@DoNotLog
-@JsonDeserialize(as = ImmutableWorkloadServerInstallConfiguration.class)
-@JsonSerialize(as = ImmutableWorkloadServerInstallConfiguration.class)
-@Value.Immutable
 public interface WorkloadServerInstallConfiguration {
```
Comment (jkozlowski) [inline atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/config/WorkloadServerInstallConfiguration.java]: I don't think internally you should be extending this, I'm assuming this is why you removed the annotations here.
Just take this as a field in the service config.

Comment (rhuffy) [reply atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/config/WorkloadServerInstallConfiguration.java]: I want to be able to do this:
```
@Value.Immutable
@ImmutablesConfigStyle
@JsonDeserialize(as = ImmutableWcWorkloadServerRuntimeConfiguration.class)
public interface WcWorkloadServerRuntimeConfiguration
        extends RuntimeConfiguration, WorkloadServerRuntimeConfiguration {}
```

Comment (rhuffy) [reply atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/config/WorkloadServerInstallConfiguration.java]: Oh I see what you mean, I can just wrap it

Comment (jkozlowski) [reply atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/config/WorkloadServerInstallConfiguration.java]: yes,

```
public interface WcWorkfloadServerRuntimeConfiguration extends RuntimeConfiguration {
  WorkloadServerRuntimeConfiguration workloadConfig();
}
```

File: atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/workflow/WorkflowFactory.java [hunk 2]
```diff
@@ -0,0 +1,254 @@
...
+import com.palantir.logsafe.logger.SafeLoggerFactory;
+import com.palantir.tritium.metrics.registry.TaggedMetricRegistry;
+import java.time.Duration;
+import java.time.Instant;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.TimeUnit;
+
+public class WorkflowFactory {
```
Comment (jeremyk-91) [inline atlasdb-workload-server/src/main/java/com/palantir/atlasdb/workload/workflow/WorkflowFactory.java]: nit: let's declare this one final?

File: atlasdb-workload-server-distribution/src/main/java/com/palantir/atlasdb/workload/DefaultWorkflowExecutorFactory.java [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+import com.palantir.atlasdb.buggify.impl.DefaultNativeSamplingSecureRandomFactory;
+import com.palantir.atlasdb.workload.workflow.WorkflowExecutorFactory;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.logger.SafeLogger;
+import com.palantir.logsafe.logger.SafeLoggerFactory;
+import io.dropwizard.lifecycle.setup.LifecycleEnvironment;
+import java.security.SecureRandom;
+import java.util.concurrent.ExecutorService;
+
+public class DefaultWorkflowExecutorFactory implements WorkflowExecutorFactory {
```
Comment (jeremyk-91) [inline atlasdb-workload-server-distribution/src/main/java/com/palantir/atlasdb/workload/DefaultWorkflowExecutorFactory.java]: as with the other one, should be `final`

**Review by jeremyk-91 — DISMISSED** [review]
👍  Minor stylistic thing to fix, but otherwise this looks good!

**Comment by jeremyk-91** [comment]
👍 thanks!

**Pull request merged**